src/vm/hints/execute_hint.rs

```diff
@@ -0,0 +1,78 @@
+use crate::vm::errors::vm_errors::VirtualMachineError;
+use crate::vm::hints::hint_utils::add_segment;
+use crate::vm::vm_core::VirtualMachine;
+
+pub fn execute_hint(vm: &mut VirtualMachine, hint_code: &[u8]) -> Result<(), VirtualMachineError> {
+    match std::str::from_utf8(hint_code) {
+        Ok("memory[ap] = segments.add()") => add_segment(vm),
+        Ok(hint_code) => Err(VirtualMachineError::UnknownHinError(String::from(
+            hint_code,
+        ))),
+        Err(_) => Err(VirtualMachineError::HintException(
+            vm.run_context.pc.clone(),
+        )),
```

Comment:
I would expect `HintException` to only be thrown when the hint itself fails. Shouldn't this be something like `BadHintEncoding` or something of the sort?